### PR TITLE
adding playstore webhook

### PR DESCRIPTION
**Pull request by husobee**
### Type of change ( select one )

- [ ] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: skus/service.go [hunk 1]
```diff
@@ -1304,6 +1304,56 @@ func (s *Service) RunStoreSignedOrderCredentialsJob(ctx context.Context) (bool,
...
+		return errors.New("notification has no purchase token")
+	}
+
+	// lookup the order based on the token as externalID
+	o, err := s.Datastore.GetOrderByExternalID(dn.SubscriptionNotification.PurchaseToken)
+	if err != nil {
+		return fmt.Errorf("failed to get order from db: %w", err)
+	}
+	// have order, now validate the receipt from the notification
+	_, err = s.validateReceipt(ctx, &o.ID, SubmitReceiptRequestV1{
```
Comment (clD11) [inline skus/service.go:1319]: could o.ID be potential nil pointer if get order by external id returns sql no rows

File: skus/datastore.go [hunk 2]
```diff
@@ -39,6 +39,8 @@ type Datastore interface {
 	SetOrderTrialDays(ctx context.Context, orderID *uuid.UUID, days int64) (*Order, error)
 	// GetOrder by ID
 	GetOrder(orderID uuid.UUID) (*Order, error)
+	// GetOrderByExternalID by the external id from the purchase vendor
+	GetOrderByExternalID(string) (*Order, error)
```
Comment (clD11) [inline skus/datastore.go:43]: possibly missing named input param

File: skus/controllers.go [hunk 3]
```diff
@@ -869,9 +869,58 @@ func VerifyCredentialV1(service *Service) handlers.AppHandler {
 func WebhookRouter(service *Service) chi.Router {
 	r := chi.NewRouter()
 	r.Method("POST", "/stripe", middleware.InstrumentHandler("HandleStripeWebhook", HandleStripeWebhook(service)))
+	r.Method("POST", "/android", middleware.InstrumentHandler("HandleAndroidWebhook", HandleAndroidWebhook(service)))
 	return r
 }
 
+// HandleAndroidWebhook is the handler for stripe checkout session webhooks
+func HandleAndroidWebhook(service *Service) handlers.AppHandler {
```
Comment (clD11) [inline skus/controllers.go:877]: Are there any tests for this. even just a happy path to test out the GetOrderByExternalID

File: skus/service.go (hunk 1 above)
Comment (husobee) [reply skus/service.go:1324]: https://github.com/brave-intl/bat-go/pull/1558/commits/1f332ef8be68974360f74e91c43396a05bb03b52

File: skus/datastore.go (hunk 2 above)
Comment (husobee) [reply skus/datastore.go]: https://github.com/brave-intl/bat-go/pull/1558/commits/1f332ef8be68974360f74e91c43396a05bb03b52

File: skus/controllers.go (hunk 3 above)
Comment (husobee) [reply skus/controllers.go:877]: https://github.com/brave-intl/bat-go/pull/1558/commits/4ec182f93df86e8e11b4b3fcf50243301414e148 adds a webhook controller test
https://github.com/brave-intl/bat-go/pull/1558/commits/1f332ef8be68974360f74e91c43396a05bb03b52  adds a get by external id test

**Review by clD11 — APPROVED** [review]

**Pull request merged**